### ISSUE TITLE
Howto multiple add_form_extra_fields in the same modelview?

### ISSUE THREAD

**Issue by vincenzodesanctis** [dpgaspar/Flask-AppBuilder#341]
Is there a way to add multiple add_form_extra_fields in the same modelview?

Insted of:
```
   add_form_extra_fields = {
                    'contact_group': AJAXSelectField('contact_group',
                    description='This will be populated with AJAX',
                    datamodel=datamodel,
                    col_name='contact_group',
                                 widget=Select2AJAXWidget(endpoint='/contactmodelview/api/column/add/contact_group')),

                    'contact_sub_group': AJAXSelectField('Extra Field2',
                    description='Extra Field description',
                    datamodel=datamodel,
                    col_name='contact_sub_group',
                    widget=Select2SlaveAJAXWidget(master_id='contact_group',
                    endpoint='/contactmodelview/api/column/add/contact_sub_group?_flt_0_contact_group_id={{ID}}'))
                    }

```

This:

```
   add_form_extra_fields = {
                    'contact_group': AJAXSelectField('contact_group',
                    description='This will be populated with AJAX',
                    datamodel=datamodel,
                    col_name='contact_group',
                                 widget=Select2AJAXWidget(endpoint='/contactmodelview/api/column/add/contact_group')),

                    'contact_sub_group': AJAXSelectField('Extra Field2',
                    description='Extra Field description',
                    datamodel=datamodel,
                    col_name='contact_sub_group',
                    widget=Select2SlaveAJAXWidget(master_id='contact_group',
                    endpoint='/contactmodelview/api/column/add/contact_sub_group?_flt_0_contact_group_id={{ID}}')),

###### another one
                    'new_one': AJAXSelectField('new_one',
                    description='This will be populated with AJAX',
                    datamodel=datamodel,
                    col_name='new_one_b',
                                 widget=Select2AJAXWidget(endpoint='/contactmodelview/api/column/add/new_one')),

                    'new_one_b': AJAXSelectField('Extra Field2',
                    description='Extra Field description',
                    datamodel=datamodel,
                    col_name='new_one_b',
                    widget=Select2SlaveAJAXWidget(master_id='contact_group',
                    endpoint='/contactmodelview/api/column/add/new_one_b?_flt_0_new_one_id={{ID}}'))


                    }
```

or this:

```
   add_form_extra_fields = {
                    'contact_group': AJAXSelectField('contact_group',
                    description='This will be populated with AJAX',
                    datamodel=datamodel,
                    col_name='contact_group',
                                 widget=Select2AJAXWidget(endpoint='/contactmodelview/api/column/add/contact_group')),

                    'contact_sub_group': AJAXSelectField('Extra Field2',
                    description='Extra Field description',
                    datamodel=datamodel,
                    col_name='contact_sub_group',
                    widget=Select2SlaveAJAXWidget(master_id='contact_group',
                    endpoint='/contactmodelview/api/column/add/contact_sub_group?_flt_0_contact_group_id={{ID}}'))
                    }


add_form_extra_fields2 = {
                    'new_one': AJAXSelectField('new_one',
                    description='This will be populated with AJAX',
                    datamodel=datamodel,
                    col_name='new_one_b',
                                 widget=Select2AJAXWidget(endpoint='/contactmodelview/api/column/add/new_one')),

                    'new_one_b': AJAXSelectField('Extra Field2',
                    description='Extra Field description',
                    datamodel=datamodel,
                    col_name='new_one_b',
                    widget=Select2SlaveAJAXWidget(master_id='contact_group',
                    endpoint='/contactmodelview/api/column/add/new_one_b?_flt_0_new_one_id={{ID}}'))


                    }
```

**Comment by dpgaspar** [comment]
Hi @vincenzodesanctis,

add_form_extra_field is a dict that will add any fields you want. It accepts:

{'FIELD_NAME', WTF_FORM_FIELD_CLASS }

Your using select2 with AJAX, is the <new_one> field defined on the model and created on the DB.
Do you get any error on this?

**Comment by vincenzodesanctis** [comment]
On my code there are 2 select2 with AJAX


```
    add_form_extra_fields = {'turno': AJAXSelectField('turno',
                             datamodel=datamodel,
                             col_name='turno',
                             widget=Select2AJAXWidget(endpoint='/contactmodelview/api/column/add/turno')),

                             'ora': AJAXSelectField('ora',
                             description='fase di debug: ora collegata con turno (relazione)',
                             datamodel=datamodel,
                             col_name='ora',
                             widget=Select2SlaveAJAXWidget(master_id='turno',
                             endpoint='/contactmodelview/api/column/add/ora?_flt_0_turno_id={{ID}}')),

#### addtional one
                             'tipologia_guasto_1': AJAXSelectField('Tipologia guasto 1',
                             datamodel=datamodel,
                             col_name='tipologia_guasto_1',
                             widget=Select2AJAXWidget(endpoint='/contactmodelview/api/column/add/tipologia_guasto_1')),

                             'dettaglio_guasto_1': AJAXSelectField('Dettaglio guasto 1',
                             description='fase di debug: ora collegata con turno (relazione)',
                             datamodel=datamodel,
                             col_name='dettaglio_guasto_1',
                             widget=Select2SlaveAJAXWidget(master_id='tipologia_guasto_1',
                             endpoint='/contactmodelview/api/column/add/dettaglio_guasto_1?_flt_0_tipologia_guasto_id={{ID}}'))
                            }
```

The first declaration override the second. If I remove the first, the second works well (both separately work well.

Is it correct:
{'FIELD_NAME', WTF_FORM_FIELD_CLASS,   'FIELD_NAME2', WTF_FORM_FIELD_CLASS2}

**Comment by dpgaspar** [comment]
Looks good! it might be a bug... let me check it.

**Comment by vincenzodesanctis** [comment]
thank you gaspar!

**Comment by dpgaspar** [comment]
Ok, i've pushed a fix (to master....), could you please try it?

**Comment by vincenzodesanctis** [comment]
Works well! Perfect! You are a very pythonist and Flaskist :+1:

**Comment by dpgaspar** [comment]
Thanks for reporting this, in fact this was a javascript problem. ;)

**Comment by dpgaspar** [comment]
Released 1.8.1 to PyPi, with this fix.

**Comment by vincenzodesanctis** [comment]
Thanks to you for the great job
